cmd/kubeadm/app/cmd/util/cmdutil.go

```diff
@@ -39,18 +39,22 @@ func SubCmdRunE(name string) func(*cobra.Command, []string) error {
 
 // ValidateExactArgNumber validates that the required top-level arguments are specified
 func ValidateExactArgNumber(args []string, supportedArgs []string) error {
+	lenSupported := len(supportedArgs)
 	validArgs := 0
 	// Disregard possible "" arguments; they are invalid
 	for _, arg := range args {
 		if len(arg) > 0 {
 			validArgs++
+			// break early for too many arguments
+			if (validArgs > lenSupported) {
+				break
+			}
 		}
 	}
 
-	if validArgs < len(supportedArgs) {
+	if validArgs < lenSupported {
 		return fmt.Errorf("missing one or more required arguments. Required arguments: %v", supportedArgs)
-	}
-	if validArgs > len(supportedArgs) {
+	} else if validArgs > lenSupported {
 		return fmt.Errorf("too many arguments, only %d argument(s) supported: %v", validArgs, supportedArgs)
```

Comment:
if you break the for loop, you will not get correct info from the error. (the validArgs is not the actual args number.)
Also, seems the validArgs and supportedArgs should change there order here.

Reply:
@xiangpengzhao 
sorry, my idea was to skip counting thousands of arguments if the number of passed arguments exceeds the number of supported arguments.

i can change the second error to read `Errorf("too many arguments. Required arguments: %v", supportedArgs)`.